packages/network-controller/src/NetworkController.ts

```diff
@@ -320,83 +333,63 @@ export class NetworkController extends BaseControllerV2<
       state.networkStatus = NetworkStatus.Unknown;
       state.networkDetails = {};
     });
-    const { rpcTarget, type, chainId, ticker } = this.state.providerConfig;
-    this.#configureProvider(type, rpcTarget, chainId, ticker);
+    const { rpcTarget, type, chainId } = this.state.providerConfig;
+    this.#configureProvider(type, rpcTarget, chainId);
     await this.lookupNetwork();
   }
 
   #registerProvider() {
     const { provider } = this.getProviderAndBlockTracker();
 
     if (provider) {
-      provider.on('error', this.#verifyNetwork.bind(this));
       this.#ethQuery = new EthQuery(provider);
     }
   }
 
-  #setupInfuraProvider(type: NetworkType) {
-    const infuraProvider = createInfuraProvider({
+  #setupInfuraProvider(type: InfuraNetworkType) {
+    const infuraProjectId = this.#infuraProjectId;
+    if (infuraProjectId === undefined) {
```

Comment:
This check was added because `createNetworkClient` wants a non-undefined `infuraProjectId` when `network` is an Infura network.